### DISCUSSION TITLE
I can't get all products on Amazon SP API

### DISCUSSION THREAD

**Discussion by ismailua** [amzn/selling-partner-api-models#3672] [General] [answered]
Hi,
I want to pull all the products in the catalog with SP API. I can pull a single product by specifying SKU, EAN or ASIN, but i want to pull all products on my catalog without specifying any of these criteria. Anyone know how to do this?
I tried some methods but it didn't work.
other suggestion;
If there is a method where I can list all the products' EAN, SKU, or ASIN codes in another way, I can access all other products in the catalog from here.

**Comment by leonardonakagawa** [comment, marked as answer]
Hello @ismailua,

Thank you for reaching out regarding how to pull all products on your catalog.

For getting all your catalog information, we recommend using the Reports API, where you can retrieve information massively from SP-API. In this particular use case, you can use the GET_MERCHANT_LISTINGS_ALL_DATA Report. This report will retrieve all the listings from your catalog. 

For more information, you can consult the [Inventory Report Type Values](https://developer-docs.amazon.com/sp-api/docs/report-type-values-inventory) and [how to use the Reports API](https://developer-docs.amazon.com/sp-api/docs/reports-api-v2021-06-30-use-case-guide)

If the information available in the documentation link provided above does not fully resolve your inquiry, please [open a support case](https://developer.amazonservices.com/support) with us.

Thanks,
Leonardo Nakagawa
Solutions Architect, Selling Partner API Developer Support

**Reply by Keaggyb123** [reply]
Hi,

Is there a way to do this for Vendor? The reports API is for sellers only.
